### ISSUE TITLE
[quotes] False positive with backtick in Lookup Types

### ISSUE THREAD

**Issue by zeckson** [typescript-eslint/typescript-eslint#2703]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
    "rules": {
      "@typescript-eslint/quotes": [
          "error",
          "backtick"
      ]
    }
}
```

```TS
interface Some {
  managerId: Manager['id'];
}
```

**Expected Result**

Should not produce any error

**Actual Result**

produces an error, and fix will cause typescript compilation to fail as back ticks are not valid Lookup Types as ts version 4.0.3.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.5.0` |
| `@typescript-eslint/parser`        | `4.5.0` |
| `TypeScript`                       | `4.0.3` |
| `ESLint`                           | `7.11.0` |
| `node`                             | `12.13.0` |

**Comment by yeonjuan** [comment]
Hi @zeckson It seems not reproduced if I used only **@typescript-eslint/quotes** rule.

 - [See online demo (only use "@typescript-eslint/quotes")](https://yeonjuan.github.io/typescript-eslint-demo/#e2NvZGU6IlxuaW50ZXJmYWNlIFNvbWUge1xuICBtYW5hZ2VySWQ6IE1hbmFnZXJbJ2lkJ107XG59IixydWxlczp7J0B0eXBlc2NyaXB0LWVzbGludC9xdW90ZXMnOlsnZXJyb3InLCdiYWNrdGljayddfSxwYXJzZXJPcHRpb25zOntlY21hRmVhdHVyZXM6e2pzeDpmYWxzZSxnbG9iYWxSZXR1cm46ZmFsc2V9LGVjbWFWZXJzaW9uOjIwMTgscHJvamVjdDpbJy4vdHNjb25maWcuanNvbiddLHNvdXJjZVR5cGU6J3NjcmlwdCd9fQ==)

It seems you use the **quotes** rule also. Could you check your configuration?
- [See online demo (use "quotes" and "@typescript-eslint/quotes") ](https://yeonjuan.github.io/typescript-eslint-demo/#e2NvZGU6IlxuaW50ZXJmYWNlIFNvbWUge1xuICBtYW5hZ2VySWQ6IE1hbmFnZXJbJ2lkJ107XG59IixydWxlczp7cXVvdGVzOlsnZXJyb3InLCdiYWNrdGljayddLCdAdHlwZXNjcmlwdC1lc2xpbnQvcXVvdGVzJzpbJ2Vycm9yJywnYmFja3RpY2snXX0scGFyc2VyT3B0aW9uczp7ZWNtYUZlYXR1cmVzOntqc3g6ZmFsc2UsZ2xvYmFsUmV0dXJuOmZhbHNlfSxlY21hVmVyc2lvbjoyMDE4LHByb2plY3Q6WycuL3RzY29uZmlnLmpzb24nXSxzb3VyY2VUeXBlOidzY3JpcHQnfX0=)

And if then you should off the **quotes** rule since it cannot treat well ts syntax.

```
{
  // note you must disable the base rule as it can report incorrect errors
  "quotes": "off",
  "@typescript-eslint/quotes": ["error", "backtic"]
}
```


cc. @bradzacher

**Comment by zeckson** [comment]
Thank you @yeonjuan! Yes, you are right it's happening only if `quotes` rule is enabled.
Since it's not obvious can we disable conflicting rules automatically in eslint?

**Comment by bradzacher** [comment]
Nope. Rules cannot influence the config. Everything must be user configured.

This is why this is documented in the readme.


https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/quotes.md#how-to-use
